pandas/tests/series/methods/test_update.py

```diff
@@ -56,3 +56,24 @@ def test_update_dtypes(self, other, dtype, expected):
         ser.update(other)
 
         tm.assert_series_equal(ser, expected)
+
+    def test_update_from_non_series(self):
+        s = Series({"a": 1, "b": 2, "c": 3, "d": 4})
+
+        d = {"b": 5, "c": np.nan}
+
+        s.update(d)
+
+        expected = Series({"a": 1, "b": 5, "c": 3, "d": 4})
+
+        tm.assert_series_equal(s, expected)
+
+        s = Series([1, 2, 3, 4])
+
+        d = [np.nan, 5, 1]
+
+        s.update(d)
+
+        expected = Series([1, 5, 1, 4])
+
+        tm.assert_series_equal(s, expected)
```

Comment:
Do you think you could remove the excess white space and parametrize over the cases tested here?

Reply:
Done!